### ISSUE TITLE
Panic: Trie Lookup error

### ISSUE THREAD

**Issue by chevdor** [paritytech/polkadot#401]
Happened on a node running polkadot 0.2.0, commit bfa142071933d56eef005a1d7640f958ba6c4eab

```
2018-07-22 13:11:43 Imported #804172 (1ac5…d337)

====================

stack backtrace:
   0:     0x556d2680e98c - backtrace::backtrace::trace::h0ff5a9bba9c3a81b

Thread '<unnamed>' panicked at 'Externalities not allowed to fail within runtime: "Trie lookup error: Database missing expected key: 0x0fbc…af11"', libcore/result.rs:945

This is a bug. Please report it at:

    https://github.com/paritytech/polkadot/issues/new
```

**Comment by gavofyork** [comment]
looks like database corruption...

**Comment by chevdor** [comment]
I tried starting the node again (docker) and it failed again, sample location, same issue.

I did a db export up to block 804100  and I am now importing. I will close this issue if the node passes block #804172.

**Comment by gavofyork** [comment]
Had the same issue on an authority node:

```
2018-07-22 15:31:53  TRACE bft  Polling BFT logic. State=(1, Start)
2018-07-22 15:31:53  TRACE bft  proposer for round 1 is 063d7787ebca768b7445dfebe7d62cbb1625ff4dba288ea34488da266dd6dca5
2018-07-22 15:31:53  TRACE bft  Polling BFT logic. State=(1, Start)
2018-07-22 15:31:53  TRACE bft  proposer for round 1 is 063d7787ebca768b7445dfebe7d62cbb1625ff4dba288ea34488da266dd6dca5
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing proposal message for round 1 from 063d?dca5
2018-07-22 15:31:53  DEBUG bft  Importing proposal for round 1
2018-07-22 15:31:53  TRACE bft  Polling BFT logic. State=(1, Start)
2018-07-22 15:31:53  DEBUG bft  evaluating block on top of parent (805330, b44c517e7a089a6f30cf5c5ea9b932e9eafca000049b52f3832ea28989ce7efe)
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Prepare(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 82c3?5cf5
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Prepare(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 063d?dca5
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Prepare(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 4de3?77e7
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Commit(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 063d?dca5
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Commit(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 82c3?5cf5
2018-07-22 15:31:53 ImportQueue TRACE sync  Block queued 805333: e7b4daa78d5a719de7b5cb1c503b1ead1e13919f5422a3b6b34fdfb6efc1e9bf
2018-07-22 15:31:53 tokio-runtime-worker-2 TRACE bft  importing vote Commit(1, 45da3a3b89f0075ed011accf1737da5f4ae03d734386492728254545173ea176) from 4de3?77e7

====================

stack backtrace:
   0:     0x5578a0fcfd0c - backtrace::backtrace::trace::h35859e07b1fccbb3

Thread '<unnamed>' panicked at 'Externalities not allowed to fail within runtime: "Trie lookup error: Database missing expected key: 0xfbf1?e0eb"', libcore/result.rs:945

This is a bug. Please report it at:

    https://github.com/paritytech/polkadot/issues/new
```

**Comment by chevdor** [comment]
What I did to *recover* is to export and import the chain. 

I don´t recommend the option as it takes really ages to import. I have a fairly good machine and importing 800k blocks took some 18+ hours.

It is currently worth doing backups of the `/db` folder.

I did import the blocks on a good machine, made a copy of the `/db` folder and brought that to my weaker VPS. Not sure if this is related to this issue but after importing up to block 804100 and restarting my node, it was saying that the best block was 540838.

Those nodes are syncing again and I did not pass blocks 800k yet. I will update once I get there (and probably do a db snapshot this time!)
